HopsanGUI/HcomHandler.cpp

```diff
@@ -6320,107 +6355,139 @@ void HcomHandler::evaluateExpression(QString expr, VariableType desiredType)
-                    return;
-                }
+        //Parse power spectrum argument
+        bool power = (powerArg == "true");
+
+        //Parse windowing function argument
+        WindowingFunctionEnumT windowingFunction;
+        if(!windowingFuncArg.isEmpty()) {
+            if(windowingFuncArg.toLower() == "hann") {
+                windowingFunction = HannWindow;
             }
-            else
-            {
-                HCOMERR(QString("Variable: %1 was not found!").arg(varName));
+            else if(windowingFuncArg.toLower() == "rectangular") {
+                windowingFunction = RectangularWindow;
+            }
+            else {
+                HCOMERR("Unknown windowing function: "+windowingFuncArg);
                 mAnsType = Undefined;
                 return;
             }
-
         }
-        else if(splitArgs.size() == 3)
-        {
-            bool power = (splitArgs[2].trimmed() == "true");
-            const QString varName = splitArgs[0].trimmed();
-            evaluateExpression(varName, DataVector);
-            SharedVectorVariableT pVar = mAnsVector;
-            if (mAnsType == DataVector)
-            {
-                const QString timeVarName = splitArgs[1].trimmed();
-                evaluateExpression(timeVarName, DataVector);
-                SharedVectorVariableT pTimeVar = mAnsVector;
-                if (mAnsType == DataVector)
-                {
-                    mAnsType = DataVector;
-                    mAnsVector = pVar->toFrequencySpectrum(pTimeVar, power);
-                    return;
-                }
-                else
-                {
-                    HCOMERR(QString("Time variable: %1 was not found!").arg(timeVarName));
-                    mAnsType = Undefined;
-                    return;
-                }
+
+        //Parse window min and max time arguments
+        double minTime, maxTime;
+        if(!minTimeArg.isEmpty()) {
+            if(maxTimeArg.isEmpty()) {
+                HCOMERR("Maximum time for windowing is required when minimum time is specified");
+                mAnsType = Undefined;
+                return;
             }
-            else
-            {
-                HCOMERR(QString("Variable: %1 was not found!").arg(varName));
+            bool ok;
+            minTime = minTimeArg.toDouble(&ok);
+            if(!ok) {
+                HCOMERR("Unknown time limit: "+minTimeArg);
+                mAnsType = Undefined;
+                return;
+            }
+            maxTime = maxTimeArg.toDouble(&ok);
+            if(!ok) {
+                HCOMERR("Unknown time limit: "+maxTimeArg);
                 mAnsType = Undefined;
                 return;
             }
         }
-        else
-        {
-            HCOMERR("Wrong number of arguments provided for fft function.\n"+mLocalFunctionDescriptions.find("fft").value().second);
-            mAnsType = Undefined;
-            return;
+
+        if(powerArg.isEmpty()) {
```

Comment:
windowing function and min max arguments can not be used if powerArg.isEmpty()
Not sure if window function is relevant. but min max time might be even in this case

Reply:
We could rearrange the number of arguments and put `power` last, but then you would have the oposite problem (i.e. need to specify windowing function if you want to use power option). I am also working on replacing the `fft()` function with two separate functions, one for energy and one for power spectrum.

Reply:
I did not look at all the code, only glanced it over so maybe I am missing something, but why cant the "windowingFunction, minTime, maxTime" arguments be included in the first case when power=true ?
And in the second case when windowfunction is not set, why cant we include min and max time ?  (just set windowing function to default value?

Reply:
There is only one function signature. If you want to use the last argument, you must provide all preceding arguments as well. It is possible to have multiple signatures, e.g. one with and one without the power argument, but it is quite complicated to distinguish between them. What if someone for example created a variable named "Hann", or a windowing function named "true"?

Reply:
I was thinking that the missing defaults could be created here, but I realize now that this is in a hcom call, so then you must have given the windowing argument if min and max is given